### DISCUSSION TITLE
Add support for web3 authentication (passwordless logins)

### DISCUSSION THREAD

**Discussion by pdarcos** [authelia/authelia#4619] [Ideas]
### Description

I'd like to be able to login without the usual web2 creds flow. Thankfully there are more and more open source libraries that can be used to login to web3 Dapps which I think would be a great addition to authelia

### Use Case

Users would be able to authenticate with just their crypto wallet in the browser instead of the usual email/password/etc combinations

### Details

This web3 model authentication greatly enhances security since there are no creds to be compromised should an attacker be able to access the backend databases.

It also makes life easier for users since they would no longer need to use the classic username/password combination.

There are many open source libraries that are used for this. One example is https://docs.walletconnect.com/2.0/introduction/auth

### Documentation

There are many open source libraries that are used for this:

https://docs.walletconnect.com/2.0/introduction/auth

https://web3auth.io/docs/overview/what-is-web3auth

And some articles:

https://www.leewayhertz.com/how-does-authentication-authorization-work-in-web3/

https://web3.hashnode.com/how-to-build-a-web3-login-with-web3js-library

https://blog.logrocket.com/build-web3-authentication-flow-react-ether-js-ceramic/

### Pre-Submission Checklist

- [X] I agree to follow the [Code of Conduct](http://www.authelia.com/code-of-conduct)
- [X] I have checked for related issues and checked the documentation

**Comment by james-d-elliott** [comment]
Unfortunately web3auth.io is using a [proprietary license model](https://web3auth.io/docs/legal/terms-and-conditions#8-our-proprietary-rights) and their product works entirely within javascript land (pretty much every other implementation of such technology runs in javascript land  currently). With the exception of an eth library there are no golang libraries I can see that support web3 authentication technologies.

I don't see the specific value this adds over a more mature specification such as Webauthn since we're not using or intending to use authelia to blockchain networks which is the key differentiator as far as I can see. I don't even think web3 authentication even has a specification yet, at least I can't see one.

**Reply by Sigri44** [reply]
You can't assign Web3 auth only to web3auth, which is a proprietary private code.

Here is good libraries :
- ethers.js : https://docs.ethers.org/v6/
- wagmi : https://wagmi.sh/react/why
- viem.sh : https://viem.sh/docs/getting-started

**Comment by james-d-elliott** [comment]
While we greatly appreciate this suggestion and want to keep it in mind I don't think it matches our design goals at the current time. We plan in the new year to add support for *optional* webauthn one-touch logins (usernameless and passwordless) in #2827, as well as platform authenticators in #2761. In addition to this multi-device registrations is about to land in #275 / #4366. After some serious thought we'll also be a OpenID RP in #2728 most likely.

These features will offer several of the benefits of this web3 proposal, without having to completely design the web3 implementation in go with only a flimsy specification to work off of.

We've decided to for the time being convert this issue to an idea discussion for the following reasons:

- This idea has __*NOT*__ been rejected/declined/etc.
- This prevents our issue list being obscure due to excessive items that we have no direct plans to implement at this stage.
- This prevents this reasonable idea from being lost if it were closed.
- This allows people to vote on the idea.

**Reply by benyanke** [reply]
Since the issue is open to discussion, I will add that passwordless email-based OTP links are certainly a feature I would use for my deployments. It's an excellent option to provide people access to private resources who may require only occasional access to the resource, and therefore may not be interested in managing an account w/ username/password, especially non-technical users.

**Reply by james-d-elliott** [reply]
We would welcome a separate discussion made regarding specifically the OTP email logins. I think I'd consider them completely separate. I also think due to lack of interest in the web3 logins we're not likely to implement that anytime soon (OTP links may be different).

**Reply by michaelscox** [reply]
Can I please weigh into this conversation.

When I am using any device that is not 100% absolutely guaranteed to be under my personal administration, operation and management fully, then I cannot trust the device. This basically means every device apart from my personal mobile and personal laptop cannot be trusted. I cannot trust the keyboard, I cannot trust the screen grabbing and I cannot trust the communications and storage interfaces. I certainly cannot trust it to not record pins, keystrokes or other biometric data. The same can be said for any operating system (including root kits) and any applications (including spyware) that may be present. 

Because of this, I usually just avoid logging into any websites using my personal credentials with the exception of my mobile and personal laptop. That is, unless the service I am trying to login to supports some form of exclusive OTP or other login method (like sending or messaging a login link or a login code via email, direct message or even QR code. All of these are great, because they can be used as the first factor and followed up with a second factor without risking compromise of my passkey or password.

In fact, I am typing this on an enterprise managed laptop that I do not trust, because I was able to login to Google using the email and device notification option, which then allowed me to use Google IAS to login to GitHub using an emailed OTP confirmation and device confirmation, which then activated GITHubs autolinking credentials (my registered email) that was verified using a followup OTP email authentication, Convoluted, yes but secure! And currently not even remotely possible with Authelia!

Please allow an option for first factor authentication THAT DOES NOT require a passkey or password.

**Comment by igun997** [comment]
Hi anyway i implement it https://wallet.enitip.com but unfortunately that just support on desktop but for send deposit that works with passkey and your passkey will saved on google. Anyway you still need any wallet for gass fee.

**Reply by anyanwudaniel** [reply]
Text me
